pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2227,3 +2227,21 @@ def test_categorical_non_unique_monotonic(n_categories):
         index=left_index,
     )
     tm.assert_frame_equal(expected, result)
+
+
+@pytest.mark.parametrize(
+    ("kwargs", "args"),
+    [
+        ({"left_on": ["b"], "right_index": True}, 1),
+        ({"left_index": True, "right_on": ["b"]}, -1),
+    ],
+)
+def test_merge_empty_right_index_left_on(kwargs, args):
```

Comment:
what is the purpose of the args argument? this is very confusing, pleas just compare the entire resulting frame.

Reply:
I reverted the list order via args to parametrize the test, but you are right. It is probably pretty confusing. I simplified it now.